src/unum/util/util.c

```diff
@@ -85,38 +85,67 @@ void util_factory_reset(void)
     log("%s: not supported on the platform\n", __func__);
 }
 
-// Populate auth_info_key. This is a unique identifier (typically
-// device serial number) that isn't as easy to guess as the MAC
-// address. It can provide extra security during initial onboardning
-// for the platforms that support it.
-#ifdef AUTH_INFO_GET_CMD
-void util_get_auth_info_key(char *auth_info_key, int max_key_len)
+// Get the command output and trim by removing trailing '\n' and '\r'
+// characters
+// cmd - Command to execute
+// output - Command output buffer
+// max_out_len - Maximum lenghth or output buffer
+void util_get_cmd_output_and_trim(char *cmd, char *output,
+                            int max_out_len)
 {
-    char *get_auth_info_key_cmd = AUTH_INFO_GET_CMD;
     int pstatus;
 
-    if(util_get_cmd_output(get_auth_info_key_cmd,
-                           auth_info_key, max_key_len, &pstatus) > 0)
+    if(util_get_cmd_output(cmd,
+                           output, max_out_len, &pstatus) > 0)
     {
         if(pstatus == 0) {
             // Chop CR/LF
             int len;
-            while((len = strlen(auth_info_key)) > 0 &&
-                  (auth_info_key[len - 1] == '\r' ||
-                   auth_info_key[len - 1] == '\n'))
+            while((len = strlen(output)) > 0 &&
+                  (output[len - 1] == '\r' ||
+                   output[len - 1] == '\n'))
             {
-                auth_info_key[len - 1] = 0;
+                output[len - 1] = 0;
             }
         } else {
-            // Some error while getting the auth info key, zero it
-            memset(auth_info_key, 0, max_key_len);
+            // Some error while getting the output, zero it
+            memset(output, 0, max_out_len);
             log("%s: Failed to get the auth info key\n", __func__);
```

Comment:
Since this is now a generic util function it should not logging.
It should return error code and the callers should log.

Reply:
Removed the log.